plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/rule/QueryGroupFeatureType.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.rule;
+
+import org.opensearch.autotagging.Attribute;
+import org.opensearch.autotagging.AutoTaggingRegistry;
+import org.opensearch.autotagging.FeatureType;
+
+import java.util.Map;
+
+/**
+ * Represents a feature type specific to the query group feature
+ * @opensearch.experimental
+ */
+public class QueryGroupFeatureType implements FeatureType {
+    /**
+     * The instance for QueryGroupFeatureType
+     */
+    public static final QueryGroupFeatureType INSTANCE = new QueryGroupFeatureType();
+    /**
+     * Name for QueryGroupFeatureType
+     */
+    public static final String NAME = "query_group";
+    private static final int MAX_ATTRIBUTE_VALUES = 10;
+    private static final int MAX_ATTRIBUTE_VALUE_LENGTH = 100;
+    private static final Map<String, Attribute> ALLOWED_ATTRIBUTES = QueryGroupAttribute.toMap();
```

Comment:
Can `QueryGroupAttribute` contain multiple attributes? Shouldn't those be separate, and this should be list of attribute implementations?

Reply:
`QueryGroupAttribute` is an limited set of values an `Attribute` can assume in context of WLM feature. Hence an enum and a list can be used interchangeably but I think it would be better to move this enum to a shared place where other plugins can use common attributes.